### DISCUSSION TITLE
DDEV Error at "Ddev debug test" - xdc

### DISCUSSION THREAD

**Discussion by tauchvogel** [ddev/ddev#3856] [General]
I am new  to DDEV. I installed it on my Raps 4 via docker/docker-compose.

My dev-site has no connection to the server, so I can not see the DDEV in the browser (the last part):

I made the "Ddev debug test" and the following was put out:
`Creating a new ddev project config in the current directory (/home/birdy/tmp/tryddevproject-20782) 
Once completed, your configuration will be written to /home/birdy/tmp/tryddevproject-20782/.ddev/config.yaml
 
Found a php codebase at /home/birdy/tmp/tryddevproject-20782 
Project type has no settings paths configured, so not creating settings file. 
Configuration complete. You may now run 'ddev start'. 
Network ddev_default created 
Starting tryddevproject-20782... 
Container ddev-ssh-agent  Started 
ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys. 
Pushed mkcert rootca certs to ddev-global-cache/mkcert 
Network ddev-tryddevproject-20782_default  Created 
Container ddev-tryddevproject-20782-dba  Started 
Container ddev-tryddevproject-20782-web  Started 
Container ddev-tryddevproject-20782-db  Started 
Container ddev-router  Started 
Project type has no settings paths configured, so not creating settings file. 
Project type has no settings paths configured, so not creating settings file. 
Successfully started tryddevproject-20782 
Project can be reached at https://tryddevproject-20782.ddev.site https://127.0.0.1:49200 
+ curl --fail -I http://tryddevproject-20782.ddev.site
HTTP/1.1 200 OK
Server: nginx/1.20.1
Date: Fri, 20 May 2022 15:36:12 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding

+ '[' 0 -ne 0 ']'
+ set +x
Thanks for running the diagnostic. It was successful.
Please provide the output of this script in a new gist at gist.github.com
Running ddev launch in 5 seconds
/usr/bin/xdg-open: 882: www-browser: not found
/usr/bin/xdg-open: 882: links2: not found
/usr/bin/xdg-open: 882: elinks: not found
/usr/bin/xdg-open: 882: links: not found
/usr/bin/xdg-open: 882: lynx: not found
/usr/bin/xdg-open: 882: w3m: not found
xdg-open: no method available for opening 'https://tryddevproject-20782.ddev.site'
Failed to run launch ; error=exit status 3

Please delete this project after debugging with 'ddev delete -Oy tryddevproject-20782'
Failed running test_ddev.sh: exit status 1
. You can run it manually with `curl -sL -O https://raw.githubusercontent.com/drud/ddev/master/cmd/ddev/cmd/scripts/test_ddev.sh && bash test_ddev.sh` `

I installed the xdc-utils, but it helped not.

What is wrong? How can I solve the problem?

**Comment by rfay** [comment]
There's nothing wrong - it works fine. But you're running it on a headless system (no GUI) and you have installed xdg-utils (so it's trying to launch a browser) and so `ddev launch` attempts to launch, but it can't.

But it looks like everything is fine to me. 

On a headless system you don't want xdg-utils. And `ddev launch` can never work there anyway.

You can see that the setup worked perfectly:
```
curl --fail -I http://tryddevproject-20782.ddev.site/
HTTP/1.1 200 OK
Server: nginx/1.20.1
Date: Fri, 20 May 2022 15:36:12 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding
```

**Comment by tauchvogel** [comment]
Okay. But there is still the problem with the server connection. It is a 502 bad gateway.

**Comment by rfay** [comment]
You'll need to say what you're trying to do. 

I imagine you're trying to run a ddev project on a remote server, which introduces a number of issues. 

You may want to look at https://ddev.readthedocs.io/en/latest/users/topics/sharing/#exposing-just-a-port-from-the-host-and-providing-a-direct-url and https://ddev.readthedocs.io/en/latest/users/alternate-uses/#alternate-uses-for-ddev-local
